src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationProperties.cs

```diff
@@ -109,6 +127,40 @@ protected void SetString(string key, string value)
             }
         }
 
+        /// <summary>
+        /// Get a parameter from the <see cref="Parameters"/> collection.
+        /// </summary>
+        /// <typeparam name="T">Parameter type.</typeparam>
+        /// <param name="key">Parameter key.</param>
+        /// <returns>Retrieved value or the default value if the property is not set.</returns>
+        public T GetParameter<T>(string key)
+        {
+            return Parameters.TryGetValue(key, out var obj) && obj is T value ? value : default;
```

Comment:
nit if you are going to right a fancy complex one liner, you might as well use => and drop the return and get rid of the extra { }'s :)